lib/dtutils.lua

```diff
@@ -0,0 +1,866 @@
+    result = os.execute("cp '" .. fromFile .. "' '" .. toFile .. "'")
+  end
+  -- if cp was not present, or if cp failed, then a pure lua solution
+  if not result then
+    local fileIn, err = io.open(fromFile, 'rb')
+    if fileIn then
+      local fileOut, errr = io.open(toFile, 'w')
+      if fileOut then
+        local content = fileIn:read(4096)
+        while content do
+          fileOut:write(content)
+          content = fileIn:read(4096)
+        end
+        result = true
+        fileIn:close()
+        fileOut:close()
+      else
+        dt.print_error("fileCopy Error: " .. errr)
+      end
+    else
+      dt.print_error("fileCopy Error: " .. err)
+    end
+  end
+  return result
+end
+
+--[[
+  NAME
+    dutils.fileMove - move a file from one directory to another
+
+  SYNOPSIS
+    result = dtutils.fileMove(fromFile, toFile)
+      fromFile - the original file
+      toFile - the new file location and name
+
+  DESCRIPTION
+    Move a file from one place to another.  Try a succession of methods from
+    builtin to operating system to a pure lua solution.
+
+  RETURN VALUE
+    result - nil on error, some value on success
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.fileMove(fromFile, toFile)
+  local success = os.rename(fromFile, toFile)
+  if not success then
+    -- an error occurred, so let's try using the operating system function
+    if dtutils.checkIfBinExists("mv") then
+      success = os.execute("mv '" .. fromFile .. "' '" .. toFile .. "'")
+    end
+    -- if the mv didn't exist or succeed, then...
+    if not success then
+      -- pure lua solution
+      success = dtutils.fileCopy(fromFile, toFile)
+      if success then
+        os.remove(fromFile)
+      else
+        dt.print_error("fileMove Error: Unable to move " .. fromFile .. " to " .. toFile .. ".  Leaving " .. fromFile .. " in place.")
+        dt.print(string.format(_("Unable to move edited file into collection. Leaving it as %s"), fromFile))
+      end
+    end
+  end
+  return success  -- nil on error, some value if success
+end
+
+--[[
+  NAME
+    dutils.updateComboboxChoices - change the list of choices in a combobox
+
+  SYNOPSIS
+    dtutils.updateComboboxChoices(combobox_widget, choice_table)
+      combobox_widget - a combobox widget
+      choice_table - a table of strings for the combobox choices
+
+  DESCRIPTION
+    Set the combobox choices to the supplied list.  Remove any extra choices from the end
+
+  RETURN VALUE
+    none
+
+  ERRORS
+
+
+  
+]]
+
+function dtutils.updateComboboxChoices(combobox, choice_table)
+  local items = #combobox
+  local choices = #choice_table
+  for i, name in ipairs(choice_table) do 
+    log.msg(log.debug, "Setting " .. i .. " to " .. name)
+    combobox[i] = name
+  end
+  if choices < items then
+    for j = items, choices + 1, -1 do
+      log.msg(log.debug, "Removing choice " .. j)
+      combobox[j] = nil
+    end
+  end
+  combobox.value = 1
+end
+
+return dtutils
```

Comment:
I'm in the process of splitting dtutils into dtutils and dtfileutils.  I'll clean up all of this while I'm doing that.